module/connection/ServerDialog.kt

```diff
@@ -62,11 +68,18 @@ object ServerDialog {
 
     private val WIDTH = 500.dp
     private val HEIGHT = 340.dp
+    private val ADDRESS_MANAGER_WIDTH = 400.dp
+    private val ADDRESS_MANAGER_HEIGHT = 500.dp
     private val appData = StudioState.appData.connection
 
+    private val state by mutableStateOf(ConnectServerForm())
+
     private class ConnectServerForm : Form.State {
-        var server: Property.Server by mutableStateOf(appData.server ?: TYPEDB)
-        var address: String by mutableStateOf(appData.address ?: "")
+        var server: Property.Server by mutableStateOf(appData.server ?: Property.Server.TYPEDB)
+        var coreAddress: String by mutableStateOf(appData.coreAddress ?: "")
+        var clusterAddresses: MutableList<String> = mutableStateListOf<String>().also {
```

Comment:
Good point. This will no longer happen. The previous behaviour seems a bit 'magical' to me, in that it goes beyond what a user expressly communicates their desire to do.